Buildings/Experimental/DHC/CentralPlants/Cooling/Subsystems/CoolingTowerWithBypass.mo

```diff
@@ -0,0 +1,391 @@
+    "Integrator time constant of the tower PID controller"
+    annotation (Dialog(enable=(controllerType == Modelica.Blocks.Types.SimpleController.PI or controllerType == Modelica.Blocks.Types.SimpleController.PID),group="Control Settings"));
+  parameter Modelica.SIunits.Time Td(
+    final min=0)=0.1
+    "Derivative time constant of the tower PID controller"
+    annotation (Dialog(enable=(controllerType == Modelica.Blocks.Types.SimpleController.PD or controllerType == Modelica.Blocks.Types.SimpleController.PID),group="Control Settings"));
+  Modelica.Blocks.Interfaces.BooleanInput on[num]
+    "On signal for cooling towers"
+    annotation (Placement(transformation(extent={{-140,20},{-100,60}})));
+  Modelica.Blocks.Interfaces.RealInput TWetBul(
+    final unit="K",
+    displayUnit="degC")
+    "Entering air wetbulb temperature"
+    annotation (Placement(transformation(extent={{-140,-40},{-100,0}})));
+  Modelica.Blocks.Interfaces.RealOutput PFan[num](
+    each final quantity="Power",
+    each final unit="W")
+    "Electric power consumed by fan"
+    annotation (Placement(transformation(extent={{100,50},{120,70}})));
+  Modelica.Blocks.Interfaces.RealOutput TLvg(
+    each final unit="K",
+    each displayUnit="degC")
+    "Leaving water temperature"
+    annotation (Placement(transformation(extent={{100,20},{120,40}})));
+  Buildings.Experimental.DHC.CentralPlants.Cooling.Subsystems.CoolingTowerParallel cooTowSys(
+    final use_inputFilter=use_inputFilter,
+    redeclare final package Medium=Medium,
+    final num=num,
+    final show_T=show_T,
+    final m_flow_small=m_flow_nominal,
+    final allowFlowReversal=allowFlowReversal,
+    final m_flow_nominal=m_flow_nominal/num,
+    final dp_nominal=dp_nominal,
+    final ratWatAir_nominal=ratWatAir_nominal,
+    final TAirInWB_nominal=TAirInWB_nominal,
+    final TWatIn_nominal=TWatIn_nominal,
+    final dT_nominal=dT_nominal,
+    final PFan_nominal=PFan_nominal,
+    final energyDynamics=energyDynamics)
+    "Cooling tower system"
+    annotation (Placement(transformation(extent={{-10,-10},{10,10}})));
+  Buildings.Fluid.Actuators.Valves.TwoWayEqualPercentage valByp(
+    redeclare final package Medium=Medium,
+    final allowFlowReversal=allowFlowReversal,
+    final m_flow_nominal=m_flow_nominal,
+    final show_T=show_T,
+    final dpValve_nominal=dpValve_nominal,
+    final dpFixed_nominal=dp_nominal,
+    final use_inputFilter=use_inputFilter)
+    "Condenser water bypass valve"
+    annotation (Placement(transformation(extent={{-10,-30},{10,-50}})));
+  Buildings.Fluid.Sensors.TemperatureTwoPort senTCWSup(
+    redeclare final package Medium=Medium,
+    final allowFlowReversal=allowFlowReversal,
+    final m_flow_nominal=m_flow_nominal,
+    final m_flow_small=m_flow_nominal,
+    final T_start=Medium.T_default)
+    "Temperature sensor"
+    annotation (Placement(transformation(extent={{60,10},{80,-10}})));
+  Modelica.Blocks.Sources.Constant TSetByPas(
```

Comment:
Please rename to `TSetByp` to be consistent with other abbreviations of bypass in this class.

Reply:
Thank you for your thoughtful suggestion, we have renamed it to keep the consistency.